src/test/java/javatests/VendorTest.java

```diff
@@ -25,58 +26,84 @@
 
 @RunWith(JUnit4.class)
 public final class VendorTest {
-    @Test
-    public void testVendorConstructorWithValidInput() {
-        // Create Vendor fields.
-        String vendorID = "vend_1";
-        String legVendID = "legVend_27";
-        int nextGenVendorID = 17;
-
-        Vendor vendor = new Vendor(vendorID, legVendID, nextGenVendorID);
+  private Vendor vendor;
+  private Account account;
+  private final String VENDOR_ID = "vend_1";
+  private final String LEGACY_VENDOR_ID = "legVend_27";
+  private final int NEXT_GEN_VENDOR_ID = 17;
+  private final String ACCOUNT_ID = "acc_12";
+  private final String ENTITY = "shopper";
+  private final String CURRENCY = "USD";
+  private final String DIRECTION = "disbursement";
+  private final String LEGACY_ACCOUNT_ID = "legAcc_53";
+  private final int NEXT_GEN_ACCOUNT_ID = 17;
+  private final String MATCHING_MODE = "straight";
+  private final String AGGREGATION_MODE = "totalAgg"; 
+     
+  /** Create a Vendor and Account object. */
+  @Before
+  public void setUp() {
+    vendor = new Vendor(VENDOR_ID, LEGACY_VENDOR_ID, NEXT_GEN_VENDOR_ID);
+    account = new Account(ACCOUNT_ID, VENDOR_ID, ENTITY, CURRENCY, DIRECTION, 
+      LEGACY_ACCOUNT_ID, NEXT_GEN_ACCOUNT_ID, MATCHING_MODE, AGGREGATION_MODE);
+  }
```

Comment:
I'm not sure if this is a real issue, but I've seen some people write that you need to have an `@After` method to clean up after every test, but I don't know if that is necessary here

Reply:
I don't know if there is anything to clean up because I am just using objects.

Reply:
sgtm, I imagine it's mostly for stuff allocated on the heap (?)